index.js

```diff
@@ -210,6 +211,30 @@ module.exports = function(version, _options) {
             }
 
             return output;
+        },
+        findAppropriateLanguage: function(language) {
+            if (languages.supportedCodes.indexOf(language) > -1) {
+                return language;
+            }
+
+            // If the language is not found, try a little harder
+            var languageCode = language.toLowerCase().split('-')[0];
+
+            var filteredSupportedLanguages = languages.supportedCodes.filter(function(locale) {
+                return locale.toLowerCase().split('-')[0] === languageCode;
+            });
+
+            if (filteredSupportedLanguages.length === 1) {
+                return filteredSupportedLanguages[0];
+            } else if (filteredSupportedLanguages.length > 1) {
+                // If more than one language is found, use the most generic version,
+                // no country code needed.
+                return filteredSupportedLanguages.sort(function(a, b) {
+                    return a.length - b.length;
+                })[0];
```

Comment:
I think we should look for matches in this order ([prior art](http://cldr.unicode.org/development/development-process/design-proposals/transform-fallback#TOC-Language-tag-fallback)), at least until we can pull in a more sophisticated library that specializes in language fallback logic:

1. Full locale code (e.g., `lng-Scpt-CC`)
1. Language code and script code (`lng-Scpt`)
1. Language code and country code (`lng-CC`)
1. Language code (`lng`)
1. Language code and any script code (`lng-Scpx`)
1. Language code and any country code (`lng-CX`)
1. English (`en`)

It’ll help if we use `languages.instructions` instead of `languages.supportedCodes`.

/cc @jcsg

Reply:
I think that sounds good. I was originally worried that condition 5 might mean that we'd serve Serbian in Latin rather than Cyrillic, but that probably would be better than just serving English for that region, especially if the voice instructions still use Serbian.

Thanks for looping me in, @1ec5 .